Ensure content-length header is only written once

paulcwarren opened the pull request:
- undertow sends all content-length headers back to the client causing browser issues
- more comments on the issue in #26330

Fixes #26330

Investigative PR

sbrannen commented:
> Investigative PR

Do you mean that we should mark this as a _Draft_ PR?

If so, please keep us posted on the status of this PR.

paulcwarren commented:
Investigative PR as in IF it is seen as a valida approach - as opposed to chasing it down to Undertow - then I am happy to iterate on the PR based on any feedback then core Spring team may have.

bclozel commented:
@paulcwarren Could you remove the changes in the test class? I think this change is a nice improvement (not setting the content length twice) but it's not fixing the core issue. I've got a change ready that should fix this inconsistency and I'll add a comment on the original issue.

paulcwarren commented:
Sure thing.  I just added that change to the test to prove the problem and TDD the fix.  Will remove and re-push.

paulcwarren committed:
Ensure content-length header is only written once

- undertow treats sends all content-length headers back to the client causing browser issues

Fixes #26330

paulcwarren commented:
Done
